index.html

```diff
@@ -1319,6 +1323,40 @@ <h3><dfn>close()</dfn> method</h3>
       <li>Return |promise|.
     </ol>
   </section>
+  <section>
+    <h3><dfn>forget()</dfn> method</h3>
+
+    <aside class="example">
+      It is posssible to voluntarily revoke a permission to a serial port that was granted by a user.
+
+      <pre class="js">
+        // Request a serial port.
+        const port = await navigator.serial.requestPort();
+
+        // Then later... revoke permission to the serial port.
+        await port.forget();
+      </pre>
+    </aside>
+
+    The {{SerialPort/forget()}} method steps are:
+
+    <ol>
+      <li>Let |promise| be [=a new promise=].
+      <li>Set [=this=].{{SerialPort/[[state]]}} to `"forgetting"`.
+      <li>
+        If [=this=].{{SerialPort/[[readable]]}} is not `null`, invoke
+        [=ReadableStream/cancel=] on [=this=].{{SerialPort/[[readable]]}}.
+      <li>
+        If [=this=].{{SerialPort/[[writable]]}} is not `null`, invoke
+        [=WritableStream/abort=] on [=this=].{{SerialPort/[[writable]]}}.
```

Comment:
Invoking `readable.cancel()` and `writable.abort()` requires the streams to be unlocked, which is basically the same thing as taking the "port must be closed to `forget()`" option I mentioned on the Chromium CL. Doing this without requiring the streams to be unlocked requires tying this algorithm in with the underlying source and sink algorithms so we can interrupt a pending operation and inject an error. I think that's a reasonable option (just add something like "if port.[[state]] becomes "forgotten" then..." to the algorithms) but it does mean we have to answer the "what if it can't be forgotten" question.

Reply:
I've added `If this.state becomes "forgotten"` as suggested to both algorithms but I'm not sure if it's the right place.